src/cmd/mesh.js

```diff
@@ -0,0 +1,340 @@
+import ParticleApi from './api';
+import { formatDeviceInfo } from './formatting';
+import { platformsById } from './constants';
+import { prompt, spin } from '../app/ui';
+
+import { openDeviceById, NotFoundError } from 'particle-usb';
+
+function canBeDeviceId(name) {
+	return /^[0-9a-f]{24}$/i.test(name);
+}
+
+export class MeshCommand {
+	constructor(settings) {
+		this._apiToken = settings.access_token;
+		this._api = new ParticleApi(settings.apiUrl, { accessToken: this._apiToken }).api;
+	}
+
+	async create(args) {
```

Comment:
We don't use `async/await` yet in the CLI since we still support Node < 10. I know we could add support by changing the Babel configuration but when we did that we had to ship the CLI with the Babel polyfill or the Babel runtime and we ran into issues with the polyfill being required multiple times in certain situations. When we last considered this we decided to stick to vanilla Promises for the time being in the CLI.

Reply:
Do you remember what was the issue with babel-runtime specifically? The CLI, including mesh commands, seems to work fine with Node 6 and the following babel config:
```
{
  "presets": ["env"],
  "plugins": ["transform-runtime", "add-module-exports"]
}
```
I'm fine with rewriting this code, I just find `await` especially helpful in tools like the CLI, where all the operations are typically done sequentially

Reply:
I don't recall the problem specifically. It was something with double includes of babel modules due to the different babel config in particle-api-js, particle-library-manager and particle-cli (if I recall correctly) crashing the CLI. Removing the need to load babel-polyfill or babel-runtime got rid of the issues. Some of these issues were reported by customers since they happened only in certain cases that we weren't testing.

I'm also a bit wary of adding a new coding style to an existing application without a plan to migrate the rest of the code base to fit the new style.

If we add async/await to the CLI, I think it should be done with `babel-plugin-transform-async-to-generator` since it brings a lot less bagage than `babel-plugin-transform-runtime` (and we don't need most of the stuff that transform-runtime gives). For example, `transform-runtime` converts `Object.assign` to a babel function which is unnecessary.

Reply:
> I'm also a bit wary of adding a new coding style to an existing application without a plan to migrate the rest of the code base to fit the new style.

very much agree.

> If we add async/await to the CLI, I think it should be done with `babel-plugin-transform-async-to-generator`

+1 here as well but also in general transpiling is kind of the worst so the less we do overall, the better imo.

that all said, Node v6 LTS hits end of life "April 2019" ([schedule](https://github.com/nodejs/Release)) and Node v8 LTS supports `async` / `await` natively ([docs](https://node.green/#ES2017-features-async-functions-await)) so fwiw

Reply:
> If we add async/await to the CLI, I think it should be done with `babel-plugin-transform-async-to-generator`

> +1 here as well but also in general transpiling is kind of the worst so the less we do overall, the better imo

If we do care about transpiling and extra stuff being pulled into the CLI, shouldn't we simply switch to `preset-env` and rely on babel to use only those transformations and polyfills which are required for the target Node version?

> I'm also a bit wary of adding a new coding style to an existing application without a plan to migrate the rest of the code base to fit the new style

`async/await` aside the CLI is already written in different styles. It's not that I want to make things worse, I just don't think that starting to use `async/await` somewhere forces us to rethink the design of the existing code. In my opinion, it's fine to use plain promises and async functions in the same codebase, moreover, you don't actually have an alternative to promises if you want to do things concurrently.

Anyway, by no means I want to start arguing over established practices in the codebase with which I'm new to. Let me know if I should rewrite the code to get this PR merged, I'm totally fine with that

Reply:
> shouldn't we simply switch to `preset-env` and rely on babel to use only those transformations and polyfills

as mentioned [above](https://github.com/particle-iot/particle-cli/pull/473#discussion_r255965192), polyfills have been problematic in the past. introducing `preset-env` as part of a larger, separate clean-up effort would probably be fine but then again dropping support for < node 8 and eliminating all pre-processing / transpiling would be even better imo.

that's all a bit of a digression though. as i would expect you'd agree, your effort here should be as focused as possible - overhauling the build system seems like it should be out of scope if only for your own sanity.

Reply:
> that's all a bit of a digression though. as i would expect you'd agree, your effort here should be as focused as possible - overhauling the build system seems like it should be out of scope

That's definitely not my intent, and I'd be happy to make a practical decision based on this discussion too. Given that polyfills are not desirable in the CLI, we have two options:
- Enable the `async-to-generator` plugin as the least invasive way to get support for `async/await` in the CLI.
- Rewrite the code to avoid using `async/await`.

I doubt the second option will take more than an hour to implement, just let me know if that's a preferred way forward

Reply:
> let me know if that's a preferred way forward

i defer to @monkbroc but my gut tells me that after wrestling w/ `npm i`, `package-lock.json` and all the additional testing that comes along with it, rewriting to avoid `async/await` will prove to be the simplest / least risky route.

fwiw, i'm also fine with enabling the `async-to-generator` plugin but there be dragons 🐲🐉 

either way, thanks for wading through all of this annoying stuff. i would like very much to get things into a better state soon 👍